File: QuranMac/Sources/QuranMac/Models/Chapter.swift
import Foundation

struct Chapter: Identifiable, Codable, Hashable {
    let id: Int
    let nameSimple: String
    let nameArabic: String
    let versesCount: Int
    let translatedName: TranslatedName?
    
    enum CodingKeys: String, CodingKey {
        case id
        case nameSimple = "name_simple"
        case nameArabic = "name_arabic"
        case versesCount = "verses_count"
        case translatedName = "translated_name"
    }
}

struct TranslatedName: Codable, Hashable {
    let name: String
    let languageName: String
    
    enum CodingKeys: String, CodingKey {
        case name
        case languageName = "language_name"
    }
}

struct ChaptersResponse: Codable {
    let chapters: [Chapter]
}

File: QuranWindows/Models/Chapter.cs
using Newtonsoft.Json;

namespace QuranWindows.Models;

public class Chapter
{
    [JsonProperty("id")]
    public int Id { get; set; }

    [JsonProperty("name_simple")]
    public string NameSimple { get; set; } = string.Empty;

    [JsonProperty("name_arabic")]
    public string NameArabic { get; set; } = string.Empty;

    [JsonProperty("verses_count")]
    public int VersesCount { get; set; }

    [JsonProperty("translated_name")]
    public TranslatedName? TranslatedName { get; set; }
}

public class TranslatedName
{
    [JsonProperty("name")]
    public string Name { get; set; } = string.Empty;

    [JsonProperty("language_name")]
    public string LanguageName { get; set; } = string.Empty;
}
